### ISSUE TITLE
Return type of _dos_getvect

### ISSUE THREAD

**Issue by TobiasKarnat** [open-watcom/open-watcom-v2#504]
Hello,

I tried to compile the following code which worked fine with the 2015er version of watcomv2.

```
static	void (__interrupt *OldKeyVect)(void);

static void
INL_StartKbd(void)
{
	IN_ClearKeysDown();

	OldKeyVect = _dos_getvect(KeyInt);
	_dos_setvect(KeyInt,INL_KeyService);
}
```

But I now get the following, how can I fix this?
> id_in.cpp(420): Error! E333: col(29) cannot convert argument to type specified in function prototype
> id_in.cpp(420): Note! N393: col(29) included from ub.cpp(4)
> id_in.cpp(420): Note! N630: col(29) source conversion type is 'void __interrupt (__far __unaligned * )( void )'
> id_in.cpp(420): Note! N631: col(29) target conversion type is 'void __interrupt (__far * )( void )'

https://github.com/TobiasKarnat/Wolf4GW

**Comment by jmalak** [comment]
Thanks for your report.
Could you mark which line is 420.
It is strange because it looks like you have somewhere used __unaligned modifier and it casused the problem. It can be related to changes in pragma aux processing which now uses full macros expanding.
How is declared INL_KeyService?

**Comment by TobiasKarnat** [comment]
Line 420 is the line which starts with _dos_setvect.

But I think it is off by one line, because if I change the declaration of
*OldKeyVect above than the target type also changes.

The declaration of INL_KeyService is
static void __interrupt
INL_KeyService(void)
{

You can find the whole source file here:
https://github.com/TobiasKarnat/Wolf4GW/blob/master/src/id_in.cpp

Am Di., 11. Dez. 2018, 06:51 hat Jiří Malák <notifications@github.com>
geschrieben:

**Comment by TobiasKarnat** [comment]
Here is the full output from OpenWatcom v2: 2015, 2017 and the current snapshot:

> E:\Wolf4GW-master\src>wmake
> Open Watcom Make Version 2.0 beta Apr  2 2015 10:33:01 (32-bit)
> Copyright (c) 2002-2015 The Open Watcom Contributors. All Rights Reserved.
> Portions Copyright (c) 1988-2002 Sybase, Inc. All Rights Reserved.
> Source code is available under the Sybase Open Watcom Public License.
> See http://www.openwatcom.org/ for details.
>         wpp386 -i=C:\WATCOM\h -bt=dos -5 -fp5 -fpi87 -ohx -j -zp1 -zq -w0 ub.cpp
>         wlink name WOLF4GW sys dos32a op el FILE ub.obj
> Open Watcom Linker Version 2.0 beta Apr  2 2015 10:32:17 (32-bit)
> Copyright (c) 2002-2015 The Open Watcom Contributors. All Rights Reserved.
> Portions Copyright (c) 1985-2002 Sybase, Inc. All Rights Reserved.
> Source code is available under the Sybase Open Watcom Public License.
> See http://www.openwatcom.org/ for details.
> loading object files
> searching libraries
> creating a DOS/32 Advanced DOS Extender (LE-style) executable
>         wpp386 -i=C:\WATCOM\h -DSPEAR -DSODPAL -bt=dos -5 -fp5 -fpi87 -ohx -j -zp1 -zq -w0 ub.cpp
>         wlink name SOD4GW sys dos32a op el FILE ub.obj
> Open Watcom Linker Version 2.0 beta Apr  2 2015 10:32:17 (32-bit)
> Copyright (c) 2002-2015 The Open Watcom Contributors. All Rights Reserved.
> Portions Copyright (c) 1985-2002 Sybase, Inc. All Rights Reserved.
> Source code is available under the Sybase Open Watcom Public License.
> See http://www.openwatcom.org/ for details.
> loading object files
> searching libraries
> creating a DOS/32 Advanced DOS Extender (LE-style) executable
> 
> E:\Wolf4GW-master\src>wmake
> Open Watcom Make Version 2.0 beta Nov  1 2017 20:57:14 (32-bit)
> Copyright (c) 2002-2017 The Open Watcom Contributors. All Rights Reserved.
> Portions Copyright (c) 1988-2002 Sybase, Inc. All Rights Reserved.
> Source code is available under the Sybase Open Watcom Public License.
> See http://www.openwatcom.org/ for details.
>         wpp386 -i=C:\WATCOM\h -bt=dos -5 -fp5 -fpi87 -ohx -j -zp1 -zq -w0 ub.cpp
> id_in.cpp(420): Error! E333: col(29) cannot convert argument to type specified in function prototype
> id_in.cpp(420): Note! N393: col(29) included from ub.cpp(4)
> id_in.cpp(420): Note! N630: col(29) source conversion type is 'void __interrupt (__far __unaligned * )( void )'
> id_in.cpp(420): Note! N631: col(29) target conversion type is 'void __interrupt (__far * )( void )'
> id_in.cpp(433): Error! E473: col(21) function argument(s) do not match those in prototype
> id_in.cpp(433): Note! N392: col(21) definition: 'void __watcall _dos_setvect( unsigned, void __interrupt (__far * )( void ))'
> id_in.cpp(433): Note! N643: col(21) cannot convert argument 2 defined in: id_in.cpp(433) (col 29)
> id_in.cpp(433): Note! N630: col(21) source conversion type is 'void __interrupt (__far __unaligned * )( void ) (lvalue)'
> id_in.cpp(433): Note! N631: col(21) target conversion type is 'void __interrupt (__far * )( void )'
> id_sd.cpp(470): Error! E473: col(29) function argument(s) do not match those in prototype
> id_sd.cpp(470): Note! N392: col(29) definition: 'void __watcall _dos_setvect( unsigned, void __interrupt (__far * )( void ))'
> id_sd.cpp(470): Note! N643: col(29) cannot convert argument 2 defined in: id_sd.cpp(470) (col 32)
> id_sd.cpp(470): Note! N630: col(29) source conversion type is 'void __interrupt (__far __unaligned * )( void ) (lvalue)'
> id_sd.cpp(470): Note! N631: col(29) target conversion type is 'void __interrupt (__far * )( void )'
> id_sd.cpp(832): Error! E333: col(31) cannot convert argument to type specified in function prototype
> id_sd.cpp(832): Note! N630: col(31) source conversion type is 'void __interrupt (__far __unaligned * )( void )'
> id_sd.cpp(832): Note! N631: col(31) target conversion type is 'void __interrupt (__far * )( void )'
> id_sd.cpp(910): Error! E473: col(21) function argument(s) do not match those in prototype
> id_sd.cpp(910): Note! N392: col(21) definition: 'void __watcall _dos_setvect( unsigned, void __interrupt (__far * )( void ))'
> id_sd.cpp(910): Note! N643: col(21) cannot convert argument 2 defined in: id_sd.cpp(910) (col 31)
> id_sd.cpp(910): Note! N630: col(21) source conversion type is 'void __interrupt (__far __unaligned * )( void ) (lvalue)'
> id_sd.cpp(910): Note! N631: col(21) target conversion type is 'void __interrupt (__far * )( void )'
> id_sd.cpp(2136): Error! E473: col(21) function argument(s) do not match those in prototype
> id_sd.cpp(2136): Note! N392: col(21) definition: 'void __watcall _dos_setvect( unsigned, void __interrupt (__far * )( void ))'
> id_sd.cpp(2136): Note! N643: col(21) cannot convert argument 2 defined in: id_sd.cpp(2136) (col 24)
> id_sd.cpp(2136): Note! N630: col(21) source conversion type is 'void __interrupt (__far __unaligned * )( void ) (lvalue)'
> id_sd.cpp(2136): Note! N631: col(21) target conversion type is 'void __interrupt (__far * )( void )'
> Error(E42): Last command making (WOLF4GW.EXE) returned a bad status
> Error(E02): Make execution terminated
> 
> 
> E:\Wolf4GW-master\src>wmake
> Open Watcom Make Version 2.0 beta Dec 12 2018 17:45:11 (32-bit)
> Copyright (c) 2002-2018 The Open Watcom Contributors. All Rights Reserved.
> Portions Copyright (c) 1988-2002 Sybase, Inc. All Rights Reserved.
> Source code is available under the Sybase Open Watcom Public License.
> See http://www.openwatcom.org/ for details.
>         wpp386 -i=C:\WATCOM\h -bt=dos -5 -fp5 -fpi87 -ohx -j -zp1 -zq -w0 ub.cpp
> id_in.cpp(420): Error! E333: col(29) cannot convert argument to type specified in function prototype
> id_in.cpp(420): Note! N393: col(29) included from ub.cpp(4)
> id_in.cpp(420): Note! N630: col(29) source conversion type is 'void __interrupt (__far __unaligned * )( void )'
> id_in.cpp(420): Note! N631: col(29) target conversion type is 'void __interrupt (__far * )( void )'
> id_in.cpp(433): Error! E473: col(21) function argument(s) do not match those in prototype
> id_in.cpp(433): Note! N392: col(21) definition: 'void __watcall _dos_setvect( unsigned, void __interrupt (__far * )( void ))'
> id_in.cpp(433): Note! N643: col(21) cannot convert argument 2 defined in: id_in.cpp(433) (col 29)
> id_in.cpp(433): Note! N630: col(21) source conversion type is 'void __interrupt (__far __unaligned * )( void ) (lvalue)'
> id_in.cpp(433): Note! N631: col(21) target conversion type is 'void __interrupt (__far * )( void )'
> id_sd.cpp(470): Error! E473: col(29) function argument(s) do not match those in prototype
> id_sd.cpp(470): Note! N392: col(29) definition: 'void __watcall _dos_setvect( unsigned, void __interrupt (__far * )( void ))'
> id_sd.cpp(470): Note! N643: col(29) cannot convert argument 2 defined in: id_sd.cpp(470) (col 32)
> id_sd.cpp(470): Note! N630: col(29) source conversion type is 'void __interrupt (__far __unaligned * )( void ) (lvalue)'
> id_sd.cpp(470): Note! N631: col(29) target conversion type is 'void __interrupt (__far * )( void )'
> id_sd.cpp(832): Error! E333: col(31) cannot convert argument to type specified in function prototype
> id_sd.cpp(832): Note! N630: col(31) source conversion type is 'void __interrupt (__far __unaligned * )( void )'
> id_sd.cpp(832): Note! N631: col(31) target conversion type is 'void __interrupt (__far * )( void )'
> id_sd.cpp(910): Error! E473: col(21) function argument(s) do not match those in prototype
> id_sd.cpp(910): Note! N392: col(21) definition: 'void __watcall _dos_setvect( unsigned, void __interrupt (__far * )( void ))'
> id_sd.cpp(910): Note! N643: col(21) cannot convert argument 2 defined in: id_sd.cpp(910) (col 31)
> id_sd.cpp(910): Note! N630: col(21) source conversion type is 'void __interrupt (__far __unaligned * )( void ) (lvalue)'
> id_sd.cpp(910): Note! N631: col(21) target conversion type is 'void __interrupt (__far * )( void )'
> id_sd.cpp(2136): Error! E473: col(21) function argument(s) do not match those in prototype
> id_sd.cpp(2136): Note! N392: col(21) definition: 'void __watcall _dos_setvect( unsigned, void __interrupt (__far * )( void ))'
> id_sd.cpp(2136): Note! N643: col(21) cannot convert argument 2 defined in: id_sd.cpp(2136) (col 24)
> id_sd.cpp(2136): Note! N630: col(21) source conversion type is 'void __interrupt (__far __unaligned * )( void ) (lvalue)'
> id_sd.cpp(2136): Note! N631: col(21) target conversion type is 'void __interrupt (__far * )( void )'
> Error(E42): Last command making (WOLF4GW.EXE) returned a bad status
> Error(E02): Make execution terminated

**Comment by jmalak** [comment]
It is fixed by commit 1f915b783ebca54044c7df6a654f1e09ed495ef4

**Comment by TobiasKarnat** [comment]
I can confirm this fixed the issue.
Thank you very much.
